### PR TITLE
Make many std.math functions CTFE-able

### PR DESCRIPTION
**Pull request by kinke**
E.g., all LLVM intrinsics currently used in `{core,std}.math`.

### CODE REVIEW

File: gen/target.cpp [hunk 1]
```diff
@@ -210,6 +210,7 @@ Expression *Target::paintAsType(Expression *e, Type *type) {
     break;
 
   case Tfloat64:
+  case Tfloat80:
```
Comment (JohanEngelen) [inline gen/target.cpp]: perhaps add a comment that the truncation to 64bit is intentional?

**Comment by JohanEngelen** [comment]
Does this needs a few simple tests?

File: gen/target.cpp (hunk 1 above)
Comment (kinke) [inline gen/target.cpp]: I'm unsure about the 2nd commit, probably needs more investigation. In any case, there's no truncation going on, these are CTFE bitcasts; the type sizes need to match, see commit msg.

Comment (JohanEngelen) [inline gen/target.cpp]: OK, but could you add that text in the code too? Thanks.
(I guess that answers the question about whether tests are needed too ;)

**Comment by kinke** [comment]
I don't think it *needs* some tests; upstream Phobos tests a few functions for CTFE availability, but none of those cover our implementations forwarding to one of these LLVM intrinsics (not CTFE-able for obvious reasons as opposed to a normal upstream implementation in D, that's why they 'need' to be detected here in `ddmd.builtin' ). A minitest wouldn't hurt either though, just to make sure this doesn't regress.

**Comment by JohanEngelen** [comment]
> A minitest wouldn't hurt either though, just to make sure this doesn't regress.

Exactly :)
(i didn't mean to test actual calculation correctness, although you could do that too if it's simple enough)

**Comment by kinke** [comment]
Added some more builtins, so that `pow(T, T|int)` works now. The test produces this on Win64:
```
abs(0.5L) = 0.5L
abs(0.5)  = 0.5
abs(0.5f) = 0.5F
fabs(0.5L) = 0.5L
fabs(0.5)  = 0.5
fabs(0.5f) = 0.5F
sqrt(0.5L) = 0.707107L
sqrt(0.5)  = 0.707107
sqrt(0.5f) = 0.707107F
sin(0.5L) = 0.479426L
sin(0.5)  = 0.479426
sin(0.5f) = 0.479426F
cos(0.5L) = 0.877583L
cos(0.5)  = 0.877583
cos(0.5f) = 0.877583F
tan(0.5L) = 0.546302L
tan(0.5)  = 0.546302L
tan(0.5f) = 0.546302L
cosh(0.5L) = 1.12763L
cosh(0.5)  = 1.12763
cosh(0.5f) = 1.12763F
asinh(0.5L) = 0.481212L
asinh(0.5)  = 0.481212
asinh(0.5f) = 0.481212F
acosh(0.5L) = -nan(ind)L
acosh(0.5)  = -nan(ind)
acosh(0.5f) = -nan(ind)F
atanh(0.5L) = 0.549306L
atanh(0.5)  = 0.549306
atanh(0.5f) = 0.549306F
ceil(0.5L) = 1.00000L
ceil(0.5)  = 1.00000
ceil(0.5f) = 1.00000F
floor(0.5L) = 0.000000L
floor(0.5)  = 0.000000
floor(0.5f) = 0.000000F
round(0.5L) = 1.00000L
round(0.5)  = 1.00000L
round(0.5f) = 1.00000L
trunc(0.5L) = 0.000000L
trunc(0.5)  = 0.000000L
trunc(0.5f) = 0.000000L
rint(0.5L) = 0.000000L
rint(0.5)  = 0.000000
rint(0.5f) = 0.000000F
nearbyint(0.5L) = 0.000000L
nearbyint(0.5)  = 0.000000L
nearbyint(0.5f) = 0.000000L
exp(0.5L) = 1.64872L
exp(0.5)  = 1.64872
exp(0.5f) = 1.64872F
exp2(0.5L) = 1.41421L
exp2(0.5)  = 1.41421L
exp2(0.5f) = 1.41421L
ldexp(0.5L, 2) = 2.00000L
ldexp(0.5,  2) = 2.00000
ldexp(0.5f, 2) = 2.00000F
log(0.5L) = -0.693147L
log(0.5)  = -0.693147L
log(0.5f) = -0.693147L
log2(0.5L) = -1L
log2(0.5)  = -1L
log2(0.5f) = -1L
log10(0.5L) = -0.30103L
log10(0.5)  = -0.30103L
log10(0.5f) = -0.30103L
pow(0.5L, -2.5L) = 5.65685L
pow(0.5,  -2.5)  = 5.65685
pow(0.5f, -2.5f) = 5.65685F
pow(0.5L, 2) = 0.25L
pow(0.5,  2) = 0.25
pow(0.5f, 2) = 0.25F
fmin(0.5L, -2.5L) = -2.5L
fmin(0.5,  -2.5)  = -2.5L
fmin(0.5f, -2.5f) = -2.5L
fmax(0.5L, -2.5L) = 0.5L
fmax(0.5,  -2.5)  = 0.5L
fmax(0.5f, -2.5f) = 0.5L
copysign(0.5L, -2.5L) = -0.5L
copysign(0.5,  -2.5)  = -0.5
copysign(0.5f, -2.5f) = -0.5F
fma(0.5L, -2.5L, 6.25L) = 5.00000L
fma(0.5,  -2.5,  6.25)  = 5.00000L
fma(0.5f, -2.5f, 6.25f) = 5.00000L
```

DMD 2.075.0 supports way less functions.

**Comment by kinke** [comment]
Okay, more work was needed to get all of them to work on hosts/targets with 80-bit reals. I'll try to get that upstreamed.

**Comment by kinke** [comment]
Travis fails with the LDC v0.17.2 host compiler because its druntime has no `core.math.yl2x[p1]`.

**Comment by kinke** [comment]
Reworked (upstream front-end patch has been merged). Requires a new prebuilt ltsmaster binary for Travis, and the Phobos patch will create a merge conflict when merging newer Phobos versions (after https://github.com/dlang/phobos/pull/5672 is merged) due to David's quadruple-precision real_t support having landed upstream already.

**Comment by kinke** [comment]
Host compiler issue solved with 0.17.5. This leaves the Phobos patch as single 'blocker' - the conflicting upstream version is included since 2.076.
